### DISCUSSION TITLE
[Other] Version Tag issue after update; database location

### DISCUSSION THREAD

**Discussion by stueja** [jonaswinkler/paperless-ng#877] [Support] [answered]
After my latest

```
docker-compose down
docker-compose pull
docker-compose up -d
```

a reload of paperless's webpage still rendered 1.3.2 as version tag (cleared cache, used different browsers, called paperless webserver directly).

After an unthoughtful

```
docker volume prune -f
```

to attempt to remove everything 1.3.2-related, now paperless's database (1400 documents) is gone.

a) which possibilities exist to re-create the database (as opposed to restore from backup, which I will try to do now)

b) which steps do I have to perform to have the `pgdata` volume _not_ as a docker volume but as a directory in my file system, just like the `media` directory? 

Thanks alot.
[docker-compose.env.txt](https://github.com/jonaswinkler/paperless-ng/files/6267651/docker-compose.env.txt)
[docker-compose.yml.txt](https://github.com/jonaswinkler/paperless-ng/files/6267652/docker-compose.yml.txt)

**Comment by shamoon** [comment]
Bummer about losing db but I think you'll need to restore, unless someone knows otherwise. Setting `pgdata` as a mount just looks like:
```
    volumes:
      - /home/docker/paperless/pgdata:/var/lib/postgresql/data
```

If you were looking for v1.4.0, I think you just need to wait longer for the new release to hit the docker registry. Also perhaps look at [watchtower](https://github.com/containrrr/watchtower) for auto-updating

**Comment by jonaswinkler** [comment, marked as answer]
> docker volume prune -f

Permanently removes all volumes that are currently not in use (i.e., mounted to a container). You'll have to restore a backup, because your database does not exist anymore. Except if you have a filesystem that creates snapshots for you, such as zfs or btrfs.

For using a directory on your file system: see above.

The version label of the latest image is set correctly.

**Reply by jonaswinkler** [reply]
There's also a delay of about an hour between the tag getting published on GitHub and the image being available on Docker Hub.

**Reply by jonaswinkler** [reply]
@stueja Additional notes that may be helpful.

When you restore an older database, it will most likely not contain all documents that were present before the prune. You can run the sanity checker after restoring, and it will identify any orphaned files (files that are not referenced by any document in the database). Remove the archived versions and thumbnails, and add the originals again. It's not too bad.

**Comment by stueja** [comment]
Thank you very much. I restored the docker volume from backup. Ran the sanity checker. It reported few orphans and a lot of checksum mismatches, e. g. checksum mismatch of document 50. I assume that this number is in the filename in the media directory, e. g. 0000050.pdf. If I open the pdf from there, it looks OK. How can I find document 50 from paperless's UI to check the metadata etc?

**Reply by jonaswinkler** [reply]
You can enter the ID in your browser. Go to the details page of any document, and look at the URL. The ID is the last bit.
